### PR TITLE
added preferences view and allow for updating delimiter (#closes 220)

### PR DESCRIPTION
**Pull request by carinedengler**
Hi,
I added the preferences view with a field to update the delimiter. I'm not sure I found all the places where the delimiter is used, I only tested that the updated delimiter is used when exporting the tokens in TEI format.

Carine

### CODE REVIEW

File: app/main/views/corpus.py [hunk 1]
```diff
@@ -260,3 +261,56 @@ def search_value_api(corpus_id, allowed_type):
             if result is not None
         ]
     )
+
+
+def _update_delimiter_token(corpus: Corpus):
```
Comment (PonteIneptique) [inline app/main/views/corpus.py:266]: In terms of design, I would rather have this as a method of Corpus:

```python
def update_delimiter_token(self, delimiter: Optional[str] = None):
```

**Comment by PonteIneptique** [comment]
Except for the comment, I think I can 95% guarantee that the delimiter is not yet used in anything else than export :)

**Comment by MrGecko** [comment]
> As a user, if I am the owner of the corpus or an administrator I want to see a link 'Preferences' in the sidebar of a corpus (I was thinking of adding it above 'Quicklinks', right under the name of the corpus)

@carinedengler I currently see the link even though I'm not an admin; I then get a 403 if I click the link.

File: app/main/views/corpus.py [hunk 2]
```diff
@@ -260,3 +261,56 @@ def search_value_api(corpus_id, allowed_type):
...
+    corpus = Corpus.query.get_or_404(corpus_id)
+    corpus_user = CorpusUser.query.filter(
+        CorpusUser.corpus_id == corpus_id,
+        CorpusUser.user_id == current_user.id
+    ).one_or_none()
+    if corpus_user:
+        is_owner = corpus_user.is_owner
+    else:
+        # admin
+        is_owner = True
```
Comment (MrGecko) [inline app/main/views/corpus.py:297]: @carinedengler not sure to understand how the admin case is handled here (not being a corpus user does not make me an admin)

File: app/main/views/corpus.py (hunk 1 above)
Comment (carinedengler) [reply app/main/views/corpus.py]: done

File: app/main/views/corpus.py (hunk 2 above)
Comment (carinedengler) [reply app/main/views/corpus.py]: > @carinedengler not sure to understand how the admin case is handled here (not being a corpus user does not make me an admin)

the reasoning was that if the user has access to the corpus, but is not part of the corpus users, it must be an admin, is there a case where this is not true?

I adapted it and added an explicit check for admin nonetheless

**Review by MrGecko — APPROVED** [review]

**Pull request merged**